docs/NotesOnConnectionAccountStates.md

```diff
@@ -0,0 +1,97 @@
+# Notes On Domain Connection and Account Login for Vircadia Web
+
+[Vircadia Web] has two connections to manage: the connection to the domain-server
+and the connection to the metaverse-server.
+The domain-server connection has to do with the display and interaction with
+the virtual world scene while the connection to the metaverse-server has to do
+with the account, access tokens, and grid management.
+
+This note discusses the states for these two connections and how they interact.
+
+# domain-server Connection
+
+The connection to the domain-server is implemented by the [Vircadia Web SDK] library.
+The connection is a little complex in that a connection to a [Vircadia Server]
+includes connections to multiple assignment-clients. The connections to the
+assignment-clients are optional (like connecting to the message assignment-client
+for text chat but not connecting to the audio assignment-client).
+
+So the connection to the domain-service can be considered as a main connection
+and then several associated sub-service connections.
+
+The persisted domain-server state will be saved as the connection URL to the domain-server,
+which sub-services are connected, and optionally an "explore" connection description
+which would give name and thumbnail information.
+
+The domain-server states will be:
+
+* disconnected
+* connecting (requires connection URL)
+* connected
+* disconnecting
+* error
+
+The same states will be used for the sub-services. Additionally, if the domain-server
+connection goes into the states "disconnected", "disconnecting", or "error", all of
+the associated sub-services will also be disconnected.
```

Comment:
The SDK assignment client modules have states:
![image](https://user-images.githubusercontent.com/7455448/133695778-002cc634-6e66-4760-8f2a-63f90329aa51.png)

Reply:
"unavailable" is a good addition.